### DISCUSSION TITLE
Big Sur update (MacOS) broke apache-spark install

### DISCUSSION THREAD

**Discussion by abelbarrera15** [Homebrew/discussions#682] [Everyday usage]
When running "brew reinstall apache-spark" or "brew install apache-spark"

I get the below:

Error: An exception occurred within a child process:
  Errno::EPERM: Operation not permitted @ rb_file_s_rename - (jars, /usr/local/Cellar/apache-spark/3.0.1/libexec/jars)

Note the dir is 7777 recursively mod'd already. And chown has been ran as per here: https://apple.stackexchange.com/questions/192227/how-to-make-files-in-usr-local-writable-for-homebrew 
  
  Is this reproducible? Is this a bug?

**Comment by SMillerDev** [comment]
What does `brew doctor` say?

**Reply by carlocab** [reply]
And `brew config`, please.

**Reply by abelbarrera15** [reply]
> What does `brew doctor` say?

Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libtcl8.6.dylib
  /usr/local/lib/libtk8.6.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/fakemysql.h
  /usr/local/include/fakepq.h
  /usr/local/include/fakesql.h
  /usr/local/include/itcl.h
  /usr/local/include/itcl2TclOO.h
  /usr/local/include/itclDecls.h
  /usr/local/include/itclInt.h
  /usr/local/include/itclIntDecls.h
  /usr/local/include/itclMigrate2TclCore.h
  /usr/local/include/itclTclIntStubsFcn.h
  /usr/local/include/mysqlStubs.h
  /usr/local/include/odbcStubs.h
  /usr/local/include/pqStubs.h
  /usr/local/include/tcl.h
  /usr/local/include/tclDecls.h
  /usr/local/include/tclOO.h
  /usr/local/include/tclOODecls.h
  /usr/local/include/tclPlatDecls.h
  /usr/local/include/tclThread.h
  /usr/local/include/tclTomMath.h
  /usr/local/include/tclTomMathDecls.h
  /usr/local/include/tdbc.h
  /usr/local/include/tdbcDecls.h
  /usr/local/include/tdbcInt.h
  /usr/local/include/tk.h
  /usr/local/include/tkDecls.h
  /usr/local/include/tkPlatDecls.h

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/tcl.pc
  /usr/local/lib/pkgconfig/tk.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libtclstub8.6.a
  /usr/local/lib/libtkstub8.6.a
  
 > And brew config, please.
 
 HOMEBREW_VERSION: 2.7.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: cd191a7f6c4c8e053336d0ea34a9886b12d06470
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 252f41dfeccc33f03543b746a56f8b63ca9c5f08
Core tap last commit: 11 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.30.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 14.0.2
macOS: 11.2-x86_64
CLT: N/A
Xcode: 12.4

**Reply by abelbarrera15** [reply]
Sooooo report as bug?

**Reply by carlocab** [reply]
It installs fine for me, so it's not clear that it's a bug. Things I'd try (though some seem unrelated to the issue you're having, they're probably worth doing anyway):

```
brew update-reset
```
Note, however:
```
❯ brew help update-reset
Usage: brew update-reset [repository ...]

Fetch and reset Homebrew and all tap repositories (or any specified
repository) using git(1) to their latest origin/HEAD.

Note: this will destroy all your uncommitted or committed changes.
```

Install the CLT with `xcode-select --install`, since you have `CLT: N/A`.

`brew config` is also showing `Java: 14.0.2`, when apache-spark needs Java 11.

**Reply by abelbarrera15** [reply]
OK I'll check out why I have such a new java and report back when I get around to it.
